### ISSUE TITLE
Failed to load zip file index

### ISSUE THREAD

**Issue by neko-para** [zen-fs/zip#2]
[resource.zip](https://github.com/zen-fs/zip/files/14905215/resource.zip)
<img width="195" alt="image" src="https://github.com/zen-fs/zip/assets/26452623/fdcaa87f-511f-484b-bda7-e8aff3ef1628">

After mounting this zip, readdir gave the following output
<img width="669" alt="image" src="https://github.com/zen-fs/zip/assets/26452623/8e6bd1fe-6e57-4b47-8263-792ca75e78be">

Seems that entries are flatten and not recognized properly.

**Comment by msk4862** [comment]
Yes I'm also facing the same issue.

Sample code that I tested:
```
const res = await fetch("source_code.zip");
const data = await res.arrayBuffer();
await configure({
  home: { backend: Zip, zipData: data },
});

fs.readdir("/home/", function (err, list) {
  console.log(err, list);
});
```

<img width="805" alt="Screenshot 2024-04-10 at 3 26 39 PM" src="https://github.com/zen-fs/zip/assets/24875366/61c40e12-aff7-4ea5-bd44-4255b8aefcfd">

**Comment by dr-vortex** [comment]
@neko-para @msk4862,

I apologize for not getting back to you sooner, I did not have the correct notification settings for this repository.

Unfortunately, I have no idea where this issue was introduced. If you have any ideas as to what is happening, I would greatly appreciate them.

In the mean time, I can start debugging and see what I come up with.

**Comment by dr-vortex** [comment]
@neko-para @msk4862,

I've been able to reproduce this issue exactly as you described. I still do not know what the cause of the issue is, so you have any insight you could share that would be nice. I will keep you updated on my debugging.

**Comment by dr-vortex** [comment]
@neko-para @msk4862,

The issue was with `safeToString` not slicing the buffer when `useUTF8` was false. This has been fixed in [v0.1.4](https://github.com/zen-fs/zip/releases/tag/v0.1.4). This fixes the issue on my end.

**Comment by neko-para** [comment]
> @neko-para @msk4862,
> 
> The issue was with `safeToString` not slicing the buffer when `useUTF8` was false. This has been fixed in [v0.1.4](https://github.com/zen-fs/zip/releases/tag/v0.1.4). This fixes the issue on my end.
> 
> Has your issue been fixed?

I'll try later, currently I've changed to using zip.js to explicitly extract and compress zip files.

**Comment by dr-vortex** [comment]
@neko-para Any updates on this?
